modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -1398,15 +1398,24 @@ type handleResponseContext struct {
 // ignoreClientGoneContext is a special context.Context type
 // intended for use when doing a RoundTrip where you don't
 // want a client disconnection to cancel the request during
-// the roundtrip. Set its done field to a Done() channel
-// of a context that doesn't get canceled when the client
-// disconnects, such as caddy.Context.Done() instead.
+// the roundtrip.
+// This can be replaced with context.WithoutCancel once
+// the minimum required required version of Go is 1.21.
```

Comment:
```suggestion
// This context clears cancellation, error, and deadline methods,
// but still allows values to pass through from its embedded
// context.
//
// TODO: This can be replaced with context.WithoutCancel once
// the minimum required required version of Go is 1.21.
```

A little clarifying

(Added the TODO just so we can find it easier later :upside_down_face: )